// @farcaster/auth-kit/dist/auth-kit.js
import { viemConnector as Vt, createAppClient as Be } from "@farcaster/auth-client";
export * from "@farcaster/auth-client";
import { jsx as C, jsxs as T, Fragment as j } from "react/jsx-runtime";
import * as I from "react";
import { createContext as Ie, useState as P, useEffect as F, useCallback as R, useContext as Me, useRef as qt, useMemo as Ae } from "react";
import { createPortal as Ne } from "react-dom";
const Pe = typeof window < "u" && (window != null && window.location) ? {
  domain: window.location.host,
  siweUri: window.location.href
} : {}, Qt = {
  relay: "https://relay.farcaster.xyz",
  version: "v1",
  ...Pe
}, Gt = Ie({
  isAuthenticated: !1,
  config: Qt,
  profile: {},
  signInMessage: {},
  onSignIn: () => {
  },
  onSignOut: () => {
  }
});
function Te({
  config: t,
  children: e
}) {
  const [n, r] = P(), [o, i] = P(!1), [a, c] = P({}), [s, f] = P({}), l = {
    ...Qt,
    ...t
  }, { relay: S, rpcUrl: h, version: v, provider: p } = l;
  F(() => {
    const g = h ? Vt({ rpcUrl: h }) : Vt(), w = Be({
      relay: S,
      ethereum: g,
      version: v
    }, p);
    r(w);
  }, [S, h, v, p]);
  const u = R((g) => {
    const { message: w, signature: m, fid: d, username: y, bio: B, displayName: b, pfpUrl: M, custody: A, verifications: N } = g;
    i(!0), c({ fid: d, username: y, bio: B, displayName: b, pfpUrl: M, custody: A, verifications: N }), f({ message: w, signature: m });
  }, []), E = () => {
    i(!1), c({}), f({});
  };
  return /* @__PURE__ */ C(
    Gt.Provider,
    {
      value: {
        appClient: n,
        isAuthenticated: o,
        profile: a,
        signInMessage: s,
        config: l,
        onSignIn: u,
        onSignOut: E
      },
      children: e
    }
  );
}
function q() {
  return Me(Gt);
}
function lt() {
  const { appClient: t } = q();
  return t;
}
function Re({
  nonce: t,
  notBefore: e,
  expirationTime: n,
  requestId: r,
  onSuccess: o,
  onError: i,
  redirectUrl: a
}) {
  const { config: c } = q(), { siweUri: s, domain: f } = c, l = lt(), [S, h] = P(!1), [v, p] = P(), [u, E] = P(), [g, w] = P(), [m, d] = P(!1), [y, B] = P(!1), [b, M] = P(), A = R(async () => {
    if (S && l && s && f && !v) {
      const _ = typeof t == "function" ? await t() : t, {
        data: mt,
        isError: vt,
        error: X
      } = await l.createChannel({
        nonce: _,
        siweUri: s,
        domain: f,
        notBefore: e,
        expirationTime: n,
        requestId: r,
        redirectUrl: a
      });
      if (vt)
        B(!0), M(X), i == null || i(X);
      else {
        const { channelToken: tt, url: Dt, nonce: Ut } = mt;
        p(tt), E(Dt), w(Ut), d(!0), o == null || o({ channelToken: tt, url: Dt, nonce: Ut });
      }
    }
  }, [
    S,
    l,
    s,
    f,
    v,
    t,
    e,
    n,
    r,
    i,
    o,
    a
  ]);
  F(() => {
    A();
  }, [A]);
  const N = R(async () => {
    h(!0);
  }, [h]), k = R(() => {
    p(void 0), E(void 0), d(!1), B(!1), M(void 0);
  }, [p, E, d, B, M]), L = R(() => {
    k(), N();
  }, [N, k]);
  return {
    connect: N,
    reconnect: L,
    reset: k,
    isSuccess: m,
    isError: y,
    error: b,
    data: { channelToken: v, url: u, nonce: g }
  };
}
function ke({
  nonce: t,
  domain: e,
  message: n,
  signature: r,
  onSuccess: o,
  onError: i
}) {
  const a = lt(), [c, s] = P(!1), [f, l] = P(!1), [S, h] = P(!1), [v, p] = P(), u = async () => {
    h(!1), l(!1), s(!1), p(void 0);
  }, E = R(async () => {
    if (a && t && e && n && r) {
      const {
        success: g,
        isError: w,
        error: m
      } = await a.verifySignInMessage({
        nonce: t,
        domain: e,
        message: n,
        signature: r
      });
      w ? (h(!0), p(m), i == null || i(m)) : (s(g), l(!0), o == null || o({ message: n, signature: r, validSignature: g }));
    }
  }, [a, t, e, n, r, o, i]);
  return F(() => {
    u(), t && e && n && r && E();
  }, [t, e, n, r, E]), {
    isSuccess: f,
    isError: S,
    error: v,
    data: { message: n, signature: r, validSignature: c }
  };
}
const Le = {
  timeout: 3e5,
  interval: 1500
};
function _e(t) {
  const e = lt(), { channelToken: n, timeout: r, interval: o, onSuccess: i, onError: a, onResponse: c } = {
    ...Le,
    ...t
  }, [s, f] = P(), [l, S] = P(!1), [h, v] = P(!1), [p, u] = P(!1), [E, g] = P(!1), [w, m] = P(), d = () => {
    f(void 0), S(!1), g(!1), v(!1), m(void 0);
  }, y = () => {
    S(!0);
  }, B = R(async () => {
    if (l && e && n) {
      u(!0);
      const {
        data: b,
        isError: M,
        error: A
      } = await e.watchStatus({
        channelToken: n,
        timeout: r,
        interval: o,
        onResponse: ({ data: N }) => {
          f(N), c == null || c(N);
        }
      });
      M ? (g(!0), u(!1), m(A), a == null || a(A)) : (v(!0), u(!1), f(b), i == null || i(b));
    }
  }, [l, e, n, r, o, i, a, c]);
  return F(() => {
    d(), n && B();
  }, [n, B]), {
    watch: y,
    isSuccess: h,
    isPolling: p,
    isError: E,
    error: w,
    data: s
  };
}
const Ze = {
  timeout: 3e5,
  interval: 1500
};
function Fe(t) {
  const e = lt(), {
    onSignIn: n,
    onSignOut: r,
    config: { domain: o, redirectUrl: i }
  } = q(), { timeout: a, interval: c, onSuccess: s, onStatusResponse: f, onError: l, ...S } = {
    ...Ze,
    ...t
  }, {
    connect: h,
    reconnect: v,
    reset: p,
    data: { channelToken: u, url: E, nonce: g },
    isSuccess: w,
    isError: m,
    error: d
  } = Re({ ...S, redirectUrl: i, onError: l }), {
    watch: y,
    isPolling: B,
    data: b,
    isError: M,
    error: A
  } = _e({
    channelToken: u,
    timeout: a,
    interval: c,
    onError: l,
    onResponse: f
  }), {
    isSuccess: N,
    data: { validSignature: k },
    isError: L,
    error: _
  } = ke({
    nonce: g,
    domain: o,
    message: b == null ? void 0 : b.message,
    signature: b == null ? void 0 : b.signature,
    onError: l
  }), mt = m || M || L, vt = d || A || _, X = R(() => {
    y();
  }, [y]), tt = R(() => {
    r(), p();
  }, [r, p]);
  return F(() => {
    N && b && k && (n(b), s == null || s(b));
  }, [N, b, k, n, s]), {
    signIn: X,
    signOut: tt,
    connect: h,
    reconnect: v,
    isConnected: w,
    isSuccess: N,
    isPolling: B,
    isError: mt,
    error: vt,
    channelToken: u,
    url: E,
    appClient: e,
    data: b,
    validSignature: k
  };
}
function Xt({
  height: t,
  fill: e
}) {
  const n = t * 1.1;
  return /* @__PURE__ */ T(
    "svg",
    {
      xmlns: "http://www.w3.org/2000/svg",
      width: n,
      height: t,
      fill: "none",
      children: [
        /* @__PURE__ */ C("title", { children: "Farcaster logo" }),
        /* @__PURE__ */ T("g", { fill: Ht[e], clipPath: "url(#a)", children: [
          /* @__PURE__ */ C("path", { d: "M3.786.05h14.156v2.824h4.025l-.844 2.825h-.714v11.427c.358 0 .65.287.65.642v.77h.13c.358 0 .649.288.649.642v.77h-7.273v-.77c0-.354.29-.642.65-.642h.13v-.77c0-.309.22-.566.512-.628l-.014-6.306c-.23-2.519-2.37-4.493-4.98-4.493-2.608 0-4.75 1.974-4.979 4.494l-.013 6.3c.346.05.772.315.772.633v.77h.13c.358 0 .65.288.65.642v.77H.15v-.77c0-.354.29-.642.649-.642h.13v-.77c0-.355.29-.642.65-.642V5.7H.863L.02 2.874h3.766V.05Z" }),
          /* @__PURE__ */ C("path", { d: "M17.942.05h.047V.003h-.047V.05ZM3.786.05V.003H3.74V.05h.047Zm14.156 2.824h-.048v.047h.048v-.047Zm4.025 0 .046.013.018-.06h-.064v.047Zm-.844 2.824v.047h.035l.01-.033-.045-.014Zm-.714 0v-.046h-.047v.046h.047Zm0 11.428h-.047v.047h.047v-.047Zm.65 1.412h-.048v.047h.047v-.047Zm.779 1.412v.047h.046v-.047h-.046Zm-7.273 0h-.048v.047h.048v-.047Zm.78-1.412v.047h.047v-.047h-.048Zm.512-1.398.01.045.038-.007v-.038h-.048Zm-.014-6.306h.048v-.004l-.048.004Zm-9.959 0-.047-.004v.004h.047Zm-.013 6.3h-.048v.041l.04.006.008-.047Zm.772 1.404h-.047v.047h.047v-.047Zm.78 1.412v.047h.047v-.047h-.048Zm-7.273 0H.102v.047H.15v-.047Zm.779-1.412v.047h.047v-.047H.93Zm.65-1.412v.047h.047v-.047h-.047Zm0-11.428h.047v-.046h-.047v.046Zm-.715 0-.045.014.01.033h.035v-.047ZM.02 2.874v-.047h-.063l.018.06.045-.013Zm3.766 0v.047h.048v-.047h-.048ZM17.942.003H3.786v.093h14.156V.003Zm.047 2.87V.05h-.095v2.824h.095Zm3.978-.046h-4.025v.094h4.025v-.094Zm-.798 2.885.844-2.825-.091-.026-.845 2.824.092.027Zm-.76.033h.714v-.093h-.714v.093Zm.048 11.381V5.698h-.095v11.428h.095Zm.649.641a.693.693 0 0 0-.697-.688v.094c.332 0 .602.266.602.595h.095Zm0 .77v-.77h-.095v.77h.095Zm.082-.045h-.13v.093h.13v-.093Zm.697.688a.692.692 0 0 0-.697-.688v.093c.333 0 .602.267.602.595h.095Zm0 .77v-.77h-.095v.77h.095Zm-3.943.047h3.896v-.094h-3.896v.094Zm-2.079 0h2.079v-.094h-2.079v.094Zm-1.298 0h1.298v-.094h-1.298v.094Zm-.048-.817v.77h.095v-.77h-.095Zm.697-.688a.693.693 0 0 0-.697.688h.095c0-.328.27-.595.602-.595v-.093Zm.13 0h-.13v.093h.13v-.093Zm-.047-.725v.77h.095v-.77h-.095Zm.55-.673a.69.69 0 0 0-.55.673h.095c0-.285.203-.524.476-.582l-.02-.09Zm-.051-6.26.014 6.306h.095l-.014-6.306h-.095Zm-4.932-4.447c2.583 0 4.704 1.956 4.932 4.452l.094-.009c-.232-2.543-2.393-4.536-5.026-4.536v.093ZM5.932 10.84c.227-2.496 2.348-4.452 4.932-4.452v-.093c-2.633 0-4.795 1.993-5.027 4.536l.095.008Zm-.014 6.295.014-6.3h-.095l-.014 6.3h.095Zm.773.633c0-.18-.12-.337-.276-.453a1.236 1.236 0 0 0-.538-.226l-.014.093c.166.025.351.1.495.207.145.108.238.241.238.38h.095Zm0 .77v-.77h-.095v.77h.095Zm.082-.045h-.13v.093h.13v-.093Zm.697.688a.693.693 0 0 0-.697-.688v.093c.332 0 .602.267.602.595h.095Zm0 .77v-.77h-.095v.77h.095Zm-1.606.047h1.558v-.094H5.864v.094Zm-.108 0h.108v-.094h-.108v.094Zm-1.97 0h1.97v-.094h-1.97v.094Zm-3.636 0h3.636v-.094H.15v.094Zm-.048-.817v.77h.095v-.77H.102Zm.697-.688a.693.693 0 0 0-.697.688h.095c0-.328.27-.595.602-.595v-.093Zm.13 0h-.13v.093h.13v-.093Zm-.048-.725v.77h.095v-.77H.881Zm.698-.688a.693.693 0 0 0-.698.688h.095c0-.328.27-.594.603-.594v-.094ZM1.53 5.7v11.427h.095V5.698h-.095Zm-.667.046h.715v-.093H.864v.093Zm-.89-2.858L.82 5.712l.09-.027-.844-2.824-.09.026Zm3.812-.06H.02v.094h3.766v-.094ZM3.74.05v2.824h.095V.05h-.095Z" })
        ] }),
        /* @__PURE__ */ C("defs", { children: /* @__PURE__ */ C("clipPath", { id: "a", children: /* @__PURE__ */ C("path", { fill: Ht[e], d: "M0 0h22v20H0z" }) }) })
      ]
    }
  );
}
const Ht = {
  purple: "#7C65C1",
  white: "#FFFFFF"
};
var De = "_1n3pr301", Ue = "_1n3pr306", Ve = "_1n3pr304", He = "_1n3pr305", xe = "_1n3pr300", Rt = "_1n3pr302", Oe = "_1n3pr303";
function Bt({ kind: t = "primary", children: e, className: n, ...r }) {
  return /* @__PURE__ */ C("button", { type: "button", className: ze[t] || n, ...r, children: e });
}
const ze = {
  primary: De,
  secondary: Rt,
  tertiary: Oe,
  reset: xe
};
function Ke({
  label: t,
  onClick: e
}) {
  return /* @__PURE__ */ T(Bt, { kind: "primary", onClick: e, children: [
    /* @__PURE__ */ C(Xt, { height: 20, fill: "white" }),
    /* @__PURE__ */ C("span", { style: { marginLeft: 9 }, children: t })
  ] });
}
function Ye(t, e) {
  F(() => {
    function n(r) {
      t.current && !t.current.contains(r.target) && e();
    }
    return document.addEventListener("mousedown", n), () => {
      document.removeEventListener("mousedown", n);
    };
  }, [t, e]);
}
var $e = "_1rbg3ua1", We = "_1rbg3ua2", je = "_1rbg3ua3", Je = "_16wbil70", qe = "_16wbil71";
function Qe({ signOut: t }) {
  return /* @__PURE__ */ C("div", { className: Je, children: /* @__PURE__ */ T(
    "button",
    {
      type: "button",
      className: Rt,
      style: {
        boxShadow: "0px 6px 12px 0 rgba(0,0,0,0.12)"
      },
      onClick: t,
      children: [
        /* @__PURE__ */ T(
          "svg",
          {
            width: "19",
            height: "20",
            viewBox: "0 0 19 20",
            fill: "none",
            xmlns: "http://www.w3.org/2000/svg",
            className: qe,
            children: [
              /* @__PURE__ */ C("title", { children: "Sign Out" }),
              /* @__PURE__ */ C(
                "path",
                {
                  d: "M11.75 7V3.25C11.75 2.65326 11.5129 2.08097 11.091 1.65901C10.669 1.23705 10.0967 1 9.5 1H3.5C2.90326 1 2.33097 1.23705 1.90901 1.65901C1.48705 2.08097 1.25 2.65326 1.25 3.25V16.75C1.25 17.3467 1.48705 17.919 1.90901 18.341C2.33097 18.7629 2.90326 19 3.5 19H9.5C10.0967 19 10.669 18.7629 11.091 18.341C11.5129 17.919 11.75 17.3467 11.75 16.75V13M14.75 13L17.75 10M17.75 10L14.75 7M17.75 10H5",
                  stroke: "black",
                  strokeWidth: "1.5",
                  strokeLinecap: "round",
                  strokeLinejoin: "round"
                }
              )
            ]
          }
        ),
        "Sign out"
      ]
    }
  ) });
}
function Ge({
  userData: t,
  signOut: e,
  hideSignOut: n
}) {
  const [r, o] = P(!1), i = qt(null);
  Ye(i, () => o(!1));
  const a = (t == null ? void 0 : t.username) ?? `!${t == null ? void 0 : t.fid}`, c = (t == null ? void 0 : t.pfpUrl) ?? "https://warpcast.com/avatar.png", s = r && !n;
  return /* @__PURE__ */ C(
    "div",
    {
      className: `fc-authkit-profile-button ${$e}`,
      ref: i,
      children: /* @__PURE__ */ T(
        "div",
        {
          style: {
            display: "flex",
            flexDirection: "column",
            alignItems: "flex-end"
          },
          children: [
            /* @__PURE__ */ T(
              "button",
              {
                type: "button",
                className: Rt,
                onClick: () => o(!r),
                children: [
                  /* @__PURE__ */ C("img", { className: We, src: c, alt: "avatar" }),
                  /* @__PURE__ */ C("span", { className: je, children: a }),
                  !n && /* @__PURE__ */ T(
                    "svg",
                    {
                      width: "24",
                      height: "24",
                      viewBox: "0 0 24 24",
                      fill: "none",
                      xmlns: "http://www.w3.org/2000/svg",
                      children: [
                        /* @__PURE__ */ C("title", { children: "Dropdown" }),
                        /* @__PURE__ */ C("g", { opacity: "0.5", children: /* @__PURE__ */ C(
                          "path",
                          {
                            fillRule: "evenodd",
                            clipRule: "evenodd",
                            d: "M6.26418 9.864C6.43293 9.69545 6.66168 9.60079 6.90018 9.60079C7.13869 9.60079 7.36743 9.69545 7.53618 9.864L12.0002 14.328L16.4642 9.864C16.5466 9.77557 16.6459 9.70465 16.7563 9.65546C16.8667 9.60627 16.9859 9.57982 17.1068 9.57769C17.2276 9.57555 17.3476 9.59778 17.4597 9.64305C17.5718 9.68831 17.6736 9.75569 17.759 9.84115C17.8445 9.92661 17.9119 10.0284 17.9571 10.1405C18.0024 10.2525 18.0246 10.3726 18.0225 10.4934C18.0204 10.6143 17.9939 10.7334 17.9447 10.8438C17.8955 10.9542 17.8246 11.0536 17.7362 11.136L12.6362 16.236C12.4674 16.4045 12.2387 16.4992 12.0002 16.4992C11.7617 16.4992 11.5329 16.4045 11.3642 16.236L6.26418 11.136C6.09564 10.9672 6.00098 10.7385 6.00098 10.5C6.00098 10.2615 6.09564 10.0327 6.26418 9.864Z",
                            fill: "black"
                          }
                        ) })
                      ]
                    }
                  )
                ]
              }
            ),
            s && /* @__PURE__ */ C(Qe, { signOut: e })
          ]
        }
      )
    }
  );
}
var D = function() {
  return D = Object.assign || function(e) {
    for (var n, r = 1, o = arguments.length; r < o; r++) {
      n = arguments[r];
      for (var i in n)
        Object.prototype.hasOwnProperty.call(n, i) && (e[i] = n[i]);
    }
    return e;
  }, D.apply(this, arguments);
};
function te(t, e) {
  var n = {};
  for (var r in t)
    Object.prototype.hasOwnProperty.call(t, r) && e.indexOf(r) < 0 && (n[r] = t[r]);
  if (t != null && typeof Object.getOwnPropertySymbols == "function")
    for (var o = 0, r = Object.getOwnPropertySymbols(t); o < r.length; o++)
      e.indexOf(r[o]) < 0 && Object.prototype.propertyIsEnumerable.call(t, r[o]) && (n[r[o]] = t[r[o]]);
  return n;
}
function Xe(t, e, n) {
  if (n || arguments.length === 2)
    for (var r = 0, o = e.length, i; r < o; r++)
      (i || !(r in e)) && (i || (i = Array.prototype.slice.call(e, 0, r)), i[r] = e[r]);
  return t.concat(i || Array.prototype.slice.call(e));
}
var it = "right-scroll-bar-position", at = "width-before-scroll-bar", tn = "with-scroll-bars-hidden", en = "--removed-body-scroll-bar-size";
function nn(t, e) {
  return typeof t == "function" ? t(e) : t && (t.current = e), t;
}
function rn(t, e) {
  var n = P(function() {
    return {
      // value
      value: t,
      // last callback
      callback: e,
      // "memoized" public interface
      facade: {
        get current() {
          return n.value;
        },
        set current(r) {
          var o = n.value;
          o !== r && (n.value = r, n.callback(r, o));
        }
      }
    };
  })[0];
  return n.callback = e, n.facade;
}
function on(t, e) {
  return rn(e || null, function(n) {
    return t.forEach(function(r) {
      return nn(r, n);
    });
  });
}
function an(t) {
  return t;
}
function cn(t, e) {
  e === void 0 && (e = an);
  var n = [], r = !1, o = {
    read: function() {
      if (r)
        throw new Error("Sidecar: could not `read` from an `assigned` medium. `read` could be used only with `useMedium`.");
      return n.length ? n[n.length - 1] : t;
    },
    useMedium: function(i) {
      var a = e(i, r);
      return n.push(a), function() {
        n = n.filter(function(c) {
          return c !== a;
        });
      };
    },
    assignSyncMedium: function(i) {
      for (r = !0; n.length; ) {
        var a = n;
        n = [], a.forEach(i);
      }
      n = {
        push: function(c) {
          return i(c);
        },
        filter: function() {
          return n;
        }
      };
    },
    assignMedium: function(i) {
      r = !0;
      var a = [];
      if (n.length) {
        var c = n;
        n = [], c.forEach(i), a = n;
      }
      var s = function() {
        var l = a;
        a = [], l.forEach(i);
      }, f = function() {
        return Promise.resolve().then(s);
      };
      f(), n = {
        push: function(l) {
          a.push(l), f();
        },
        filter: function(l) {
          return a = a.filter(l), n;
        }
      };
    }
  };
  return o;
}
function sn(t) {
  t === void 0 && (t = {});
  var e = cn(null);
  return e.options = D({ async: !0, ssr: !1 }, t), e;
}
var ee = function(t) {
  var e = t.sideCar, n = te(t, ["sideCar"]);
  if (!e)
    throw new Error("Sidecar: please provide `sideCar` property to import the right car");
  var r = e.read();
  if (!r)
    throw new Error("Sidecar medium not found");
  return I.createElement(r, D({}, n));
};
ee.isSideCarExport = !0;
function ln(t, e) {
  return t.useMedium(e), ee;
}
var ne = sn(), wt = function() {
}, ut = I.forwardRef(function(t, e) {
  var n = I.useRef(null), r = I.useState({
    onScrollCapture: wt,
    onWheelCapture: wt,
    onTouchMoveCapture: wt
  }), o = r[0], i = r[1], a = t.forwardProps, c = t.children, s = t.className, f = t.removeScrollBar, l = t.enabled, S = t.shards, h = t.sideCar, v = t.noIsolation, p = t.inert, u = t.allowPinchZoom, E = t.as, g = E === void 0 ? "div" : E, w = t.gapMode, m = te(t, ["forwardProps", "children", "className", "removeScrollBar", "enabled", "shards", "sideCar", "noIsolation", "inert", "allowPinchZoom", "as", "gapMode"]), d = h, y = on([n, e]), B = D(D({}, m), o);
  return I.createElement(
    I.Fragment,
    null,
    l && I.createElement(d, { sideCar: ne, removeScrollBar: f, shards: S, noIsolation: v, inert: p, setCallbacks: i, allowPinchZoom: !!u, lockRef: n, gapMode: w }),
    a ? I.cloneElement(I.Children.only(c), D(D({}, B), { ref: y })) : I.createElement(g, D({}, B, { className: s, ref: y }), c)
  );
});
ut.defaultProps = {
  enabled: !0,
  removeScrollBar: !0,
  inert: !1
};
ut.classNames = {
  fullWidth: at,
  zeroRight: it
};
var un = function() {
  if (typeof __webpack_nonce__ < "u")
    return __webpack_nonce__;
};
function fn() {
  if (!document)
    return null;
  var t = document.createElement("style");
  t.type = "text/css";
  var e = un();
  return e && t.setAttribute("nonce", e), t;
}
function dn(t, e) {
  t.styleSheet ? t.styleSheet.cssText = e : t.appendChild(document.createTextNode(e));
}
function hn(t) {
  var e = document.head || document.getElementsByTagName("head")[0];
  e.appendChild(t);
}
var gn = function() {
  var t = 0, e = null;
  return {
    add: function(n) {
      t == 0 && (e = fn()) && (dn(e, n), hn(e)), t++;
    },
    remove: function() {
      t--, !t && e && (e.parentNode && e.parentNode.removeChild(e), e = null);
    }
  };
}, mn = function() {
  var t = gn();
  return function(e, n) {
    I.useEffect(function() {
      return t.add(e), function() {
        t.remove();
      };
    }, [e && n]);
  };
}, re = function() {
  var t = mn(), e = function(n) {
    var r = n.styles, o = n.dynamic;
    return t(r, o), null;
  };
  return e;
}, vn = {
  left: 0,
  top: 0,
  right: 0,
  gap: 0
}, pt = function(t) {
  return parseInt(t || "", 10) || 0;
}, wn = function(t) {
  var e = window.getComputedStyle(document.body), n = e[t === "padding" ? "paddingLeft" : "marginLeft"], r = e[t === "padding" ? "paddingTop" : "marginTop"], o = e[t === "padding" ? "paddingRight" : "marginRight"];
  return [pt(n), pt(r), pt(o)];
}, pn = function(t) {
  if (t === void 0 && (t = "margin"), typeof window > "u")
    return vn;
  var e = wn(t), n = document.documentElement.clientWidth, r = window.innerWidth;
  return {
    left: e[0],
    top: e[1],
    right: e[2],
    gap: Math.max(0, r - n + e[2] - e[0])
  };
}, yn = re(), Cn = function(t, e, n, r) {
  var o = t.left, i = t.top, a = t.right, c = t.gap;
  return n === void 0 && (n = "margin"), `
  .`.concat(tn, ` {
   overflow: hidden `).concat(r, `;
   padding-right: `).concat(c, "px ").concat(r, `;
  }
  body {
    overflow: hidden `).concat(r, `;
    overscroll-behavior: contain;
    `).concat([
    e && "position: relative ".concat(r, ";"),
    n === "margin" && `
    padding-left: `.concat(o, `px;
    padding-top: `).concat(i, `px;
    padding-right: `).concat(a, `px;
    margin-left:0;
    margin-top:0;
    margin-right: `).concat(c, "px ").concat(r, `;
    `),
    n === "padding" && "padding-right: ".concat(c, "px ").concat(r, ";")
  ].filter(Boolean).join(""), `
  }
  
  .`).concat(it, ` {
    right: `).concat(c, "px ").concat(r, `;
  }
  
  .`).concat(at, ` {
    margin-right: `).concat(c, "px ").concat(r, `;
  }
  
  .`).concat(it, " .").concat(it, ` {
    right: 0 `).concat(r, `;
  }
  
  .`).concat(at, " .").concat(at, ` {
    margin-right: 0 `).concat(r, `;
  }
  
  body {
    `).concat(en, ": ").concat(c, `px;
  }
`);
}, En = function(t) {
  var e = t.noRelative, n = t.noImportant, r = t.gapMode, o = r === void 0 ? "margin" : r, i = I.useMemo(function() {
    return pn(o);
  }, [o]);
  return I.createElement(yn, { styles: Cn(i, !e, o, n ? "" : "!important") });
}, It = !1;
if (typeof window < "u")
  try {
    var et = Object.defineProperty({}, "passive", {
      get: function() {
        return It = !0, !0;
      }
    });
    window.addEventListener("test", et, et), window.removeEventListener("test", et, et);
  } catch {
    It = !1;
  }
var x = It ? { passive: !1 } : !1, Sn = function(t) {
  return t.tagName === "TEXTAREA";
}, oe = function(t, e) {
  var n = window.getComputedStyle(t);
  return (
    // not-not-scrollable
    n[e] !== "hidden" && // contains scroll inside self
    !(n.overflowY === n.overflowX && !Sn(t) && n[e] === "visible")
  );
}, bn = function(t) {
  return oe(t, "overflowY");
}, Bn = function(t) {
  return oe(t, "overflowX");
}, xt = function(t, e) {
  var n = e.ownerDocument, r = e;
  do {
    typeof ShadowRoot < "u" && r instanceof ShadowRoot && (r = r.host);
    var o = ie(t, r);
    if (o) {
      var i = ae(t, r), a = i[1], c = i[2];
      if (a > c)
        return !0;
    }
    r = r.parentNode;
  } while (r && r !== n.body);
  return !1;
}, In = function(t) {
  var e = t.scrollTop, n = t.scrollHeight, r = t.clientHeight;
  return [
    e,
    n,
    r
  ];
}, Mn = function(t) {
  var e = t.scrollLeft, n = t.scrollWidth, r = t.clientWidth;
  return [
    e,
    n,
    r
  ];
}, ie = function(t, e) {
  return t === "v" ? bn(e) : Bn(e);
}, ae = function(t, e) {
  return t === "v" ? In(e) : Mn(e);
}, An = function(t, e) {
  return t === "h" && e === "rtl" ? -1 : 1;
}, Nn = function(t, e, n, r, o) {
  var i = An(t, window.getComputedStyle(e).direction), a = i * r, c = n.target, s = e.contains(c), f = !1, l = a > 0, S = 0, h = 0;
  do {
    var v = ae(t, c), p = v[0], u = v[1], E = v[2], g = u - E - i * p;
    (p || g) && ie(t, c) && (S += g, h += p), c instanceof ShadowRoot ? c = c.host : c = c.parentNode;
  } while (
    // portaled content
    !s && c !== document.body || // self content
    s && (e.contains(c) || e === c)
  );
  return (l && (o && Math.abs(S) < 1 || !o && a > S) || !l && (o && Math.abs(h) < 1 || !o && -a > h)) && (f = !0), f;
}, nt = function(t) {
  return "changedTouches" in t ? [t.changedTouches[0].clientX, t.changedTouches[0].clientY] : [0, 0];
}, Ot = function(t) {
  return [t.deltaX, t.deltaY];
}, zt = function(t) {
  return t && "current" in t ? t.current : t;
}, Pn = function(t, e) {
  return t[0] === e[0] && t[1] === e[1];
}, Tn = function(t) {
  return `
  .block-interactivity-`.concat(t, ` {pointer-events: none;}
  .allow-interactivity-`).concat(t, ` {pointer-events: all;}
`);
}, Rn = 0, O = [];
function kn(t) {
  var e = I.useRef([]), n = I.useRef([0, 0]), r = I.useRef(), o = I.useState(Rn++)[0], i = I.useState(re)[0], a = I.useRef(t);
  I.useEffect(function() {
    a.current = t;
  }, [t]), I.useEffect(function() {
    if (t.inert) {
      document.body.classList.add("block-interactivity-".concat(o));
      var u = Xe([t.lockRef.current], (t.shards || []).map(zt), !0).filter(Boolean);
      return u.forEach(function(E) {
        return E.classList.add("allow-interactivity-".concat(o));
      }), function() {
        document.body.classList.remove("block-interactivity-".concat(o)), u.forEach(function(E) {
          return E.classList.remove("allow-interactivity-".concat(o));
        });
      };
    }
  }, [t.inert, t.lockRef.current, t.shards]);
  var c = I.useCallback(function(u, E) {
    if ("touches" in u && u.touches.length === 2)
      return !a.current.allowPinchZoom;
    var g = nt(u), w = n.current, m = "deltaX" in u ? u.deltaX : w[0] - g[0], d = "deltaY" in u ? u.deltaY : w[1] - g[1], y, B = u.target, b = Math.abs(m) > Math.abs(d) ? "h" : "v";
    if ("touches" in u && b === "h" && B.type === "range")
      return !1;
    var M = xt(b, B);
    if (!M)
      return !0;
    if (M ? y = b : (y = b === "v" ? "h" : "v", M = xt(b, B)), !M)
      return !1;
    if (!r.current && "changedTouches" in u && (m || d) && (r.current = y), !y)
      return !0;
    var A = r.current || y;
    return Nn(A, E, u, A === "h" ? m : d, !0);
  }, []), s = I.useCallback(function(u) {
    var E = u;
    if (!(!O.length || O[O.length - 1] !== i)) {
      var g = "deltaY" in E ? Ot(E) : nt(E), w = e.current.filter(function(y) {
        return y.name === E.type && (y.target === E.target || E.target === y.shadowParent) && Pn(y.delta, g);
      })[0];
      if (w && w.should) {
        E.cancelable && E.preventDefault();
        return;
      }
      if (!w) {
        var m = (a.current.shards || []).map(zt).filter(Boolean).filter(function(y) {
          return y.contains(E.target);
        }), d = m.length > 0 ? c(E, m[0]) : !a.current.noIsolation;
        d && E.cancelable && E.preventDefault();
      }
    }
  }, []), f = I.useCallback(function(u, E, g, w) {
    var m = { name: u, delta: E, target: g, should: w, shadowParent: Ln(g) };
    e.current.push(m), setTimeout(function() {
      e.current = e.current.filter(function(d) {
        return d !== m;
      });
    }, 1);
  }, []), l = I.useCallback(function(u) {
    n.current = nt(u), r.current = void 0;
  }, []), S = I.useCallback(function(u) {
    f(u.type, Ot(u), u.target, c(u, t.lockRef.current));
  }, []), h = I.useCallback(function(u) {
    f(u.type, nt(u), u.target, c(u, t.lockRef.current));
  }, []);
  I.useEffect(function() {
    return O.push(i), t.setCallbacks({
      onScrollCapture: S,
      onWheelCapture: S,
      onTouchMoveCapture: h
    }), document.addEventListener("wheel", s, x), document.addEventListener("touchmove", s, x), document.addEventListener("touchstart", l, x), function() {
      O = O.filter(function(u) {
        return u !== i;
      }), document.removeEventListener("wheel", s, x), document.removeEventListener("touchmove", s, x), document.removeEventListener("touchstart", l, x);
    };
  }, []);
  var v = t.removeScrollBar, p = t.inert;
  return I.createElement(
    I.Fragment,
    null,
    p ? I.createElement(i, { styles: Tn(o) }) : null,
    v ? I.createElement(En, { gapMode: t.gapMode }) : null
  );
}
function Ln(t) {
  for (var e = null; t !== null; )
    t instanceof ShadowRoot && (e = t.host, t = t.host), t = t.parentNode;
  return e;
}
const _n = ln(ne, kn);
var ce = I.forwardRef(function(t, e) {
  return I.createElement(ut, D({}, t, { ref: e, sideCar: _n }));
});
ce.classNames = ut.classNames;
const Zn = ce;
var Fn = "nao3n43", Dn = "nao3n42";
const Kt = (t, e) => {
  const n = t.querySelectorAll("button:not(:disabled), a[href]");
  n.length !== 0 && n[e === "end" ? n.length - 1 : 0].focus();
};
function Un(t) {
  const e = qt(null);
  return F(() => {
    const n = document.activeElement;
    return () => {
      var r;
      (r = n.focus) == null || r.call(n);
    };
  }, []), F(() => {
    if (e.current) {
      const n = e.current.querySelector("[data-auto-focus]");
      n ? n.focus() : e.current.focus();
    }
  }, []), /* @__PURE__ */ T(j, { children: [
    /* @__PURE__ */ C(
      "div",
      {
        onFocus: R(() => e.current && Kt(e.current, "end"), []),
        tabIndex: 0
      }
    ),
    /* @__PURE__ */ C("div", { ref: e, style: { outline: "none" }, tabIndex: -1, ...t }),
    /* @__PURE__ */ C(
      "div",
      {
        onFocus: R(() => e.current && Kt(e.current, "start"), []),
        tabIndex: 0
      }
    )
  ] });
}
function Vn() {
  return typeof navigator < "u" && /android/i.test(navigator.userAgent);
}
function Hn() {
  return typeof navigator < "u" && /iPhone|iPod/.test(navigator.userAgent);
}
function xn() {
  return typeof navigator < "u" && (/iPad/.test(navigator.userAgent) || navigator.platform === "MacIntel" && navigator.maxTouchPoints > 1);
}
function On() {
  return Hn() || xn();
}
function Mt() {
  return Vn() || On();
}
const zn = (t) => t.stopPropagation();
function Kn({ children: t, onClose: e, open: n, titleId: r }) {
  F(() => {
    const c = (s) => n && s.key === "Escape" && e();
    return document.addEventListener("keydown", c), () => document.removeEventListener("keydown", c);
  }, [n, e]);
  const [o, i] = P(!0);
  F(() => {
    i(
      getComputedStyle(window.document.body).overflow !== "hidden"
    );
  }, []);
  const a = R(() => e(), [e]);
  return /* @__PURE__ */ C(j, { children: n ? Ne(
    /* @__PURE__ */ C(Zn, { enabled: o, children: /* @__PURE__ */ C(
      "div",
      {
        style: {
          alignItems: Mt() ? "flex-end" : "center",
          position: "fixed"
        },
        "aria-labelledby": r,
        "aria-modal": !0,
        className: Dn,
        onClick: a,
        role: "dialog",
        children: /* @__PURE__ */ C(
          Un,
          {
            className: Fn,
            onClick: zn,
            role: "document",
            children: t
          }
        )
      }
    ) }),
    document.body
  ) : null });
}
var Yn = "_1sj0tk80", Yt = "_1sj0tk82", $t = "_1sj0tk81", Q = {}, $n = function() {
  return typeof Promise == "function" && Promise.prototype && Promise.prototype.then;
}, se = {}, Z = {};
let kt;
const Wn = [
  0,
  // Not used
  26,
  44,
  70,
  100,
  134,
  172,
  196,
  242,
  292,
  346,
  404,
  466,
  532,
  581,
  655,
  733,
  815,
  901,
  991,
  1085,
  1156,
  1258,
  1364,
  1474,
  1588,
  1706,
  1828,
  1921,
  2051,
  2185,
  2323,
  2465,
  2611,
  2761,
  2876,
  3034,
  3196,
  3362,
  3532,
  3706
];
Z.getSymbolSize = function(e) {
  if (!e)
    throw new Error('"version" cannot be null or undefined');
  if (e < 1 || e > 40)
    throw new Error('"version" should be in range from 1 to 40');
  return e * 4 + 17;
};
Z.getSymbolTotalCodewords = function(e) {
  return Wn[e];
};
Z.getBCHDigit = function(t) {
  let e = 0;
  for (; t !== 0; )
    e++, t >>>= 1;
  return e;
};
Z.setToSJISFunction = function(e) {
  if (typeof e != "function")
    throw new Error('"toSJISFunc" is not a valid function.');
  kt = e;
};
Z.isKanjiModeEnabled = function() {
  return typeof kt < "u";
};
Z.toSJIS = function(e) {
  return kt(e);
};
var ft = {};
(function(t) {
  t.L = { bit: 1 }, t.M = { bit: 0 }, t.Q = { bit: 3 }, t.H = { bit: 2 };
  function e(n) {
    if (typeof n != "string")
      throw new Error("Param is not a string");
    switch (n.toLowerCase()) {
      case "l":
      case "low":
        return t.L;
      case "m":
      case "medium":
        return t.M;
      case "q":
      case "quartile":
        return t.Q;
      case "h":
      case "high":
        return t.H;
      default:
        throw new Error("Unknown EC Level: " + n);
    }
  }
  t.isValid = function(r) {
    return r && typeof r.bit < "u" && r.bit >= 0 && r.bit < 4;
  }, t.from = function(r, o) {
    if (t.isValid(r))
      return r;
    try {
      return e(r);
    } catch {
      return o;
    }
  };
})(ft);
function le() {
  this.buffer = [], this.length = 0;
}
le.prototype = {
  get: function(t) {
    const e = Math.floor(t / 8);
    return (this.buffer[e] >>> 7 - t % 8 & 1) === 1;
  },
  put: function(t, e) {
    for (let n = 0; n < e; n++)
      this.putBit((t >>> e - n - 1 & 1) === 1);
  },
  getLengthInBits: function() {
    return this.length;
  },
  putBit: function(t) {
    const e = Math.floor(this.length / 8);
    this.buffer.length <= e && this.buffer.push(0), t && (this.buffer[e] |= 128 >>> this.length % 8), this.length++;
  }
};
var jn = le;
function G(t) {
  if (!t || t < 1)
    throw new Error("BitMatrix size must be defined and greater than 0");
  this.size = t, this.data = new Uint8Array(t * t), this.reservedBit = new Uint8Array(t * t);
}
G.prototype.set = function(t, e, n, r) {
  const o = t * this.size + e;
  this.data[o] = n, r && (this.reservedBit[o] = !0);
};
G.prototype.get = function(t, e) {
  return this.data[t * this.size + e];
};
G.prototype.xor = function(t, e, n) {
  this.data[t * this.size + e] ^= n;
};
G.prototype.isReserved = function(t, e) {
  return this.reservedBit[t * this.size + e];
};
var Jn = G, ue = {};
(function(t) {
  const e = Z.getSymbolSize;
  t.getRowColCoords = function(r) {
    if (r === 1)
      return [];
    const o = Math.floor(r / 7) + 2, i = e(r), a = i === 145 ? 26 : Math.ceil((i - 13) / (2 * o - 2)) * 2, c = [i - 7];
    for (let s = 1; s < o - 1; s++)
      c[s] = c[s - 1] - a;
    return c.push(6), c.reverse();
  }, t.getPositions = function(r) {
    const o = [], i = t.getRowColCoords(r), a = i.length;
    for (let c = 0; c < a; c++)
      for (let s = 0; s < a; s++)
        c === 0 && s === 0 || // top-left
        c === 0 && s === a - 1 || // bottom-left
        c === a - 1 && s === 0 || o.push([i[c], i[s]]);
    return o;
  };
})(ue);
var fe = {};
const qn = Z.getSymbolSize, Wt = 7;
fe.getPositions = function(e) {
  const n = qn(e);
  return [
    // top-left
    [0, 0],
    // top-right
    [n - Wt, 0],
    // bottom-left
    [0, n - Wt]
  ];
};
var de = {};
(function(t) {
  t.Patterns = {
    PATTERN000: 0,
    PATTERN001: 1,
    PATTERN010: 2,
    PATTERN011: 3,
    PATTERN100: 4,
    PATTERN101: 5,
    PATTERN110: 6,
    PATTERN111: 7
  };
  const e = {
    N1: 3,
    N2: 3,
    N3: 40,
    N4: 10
  };
  t.isValid = function(o) {
    return o != null && o !== "" && !isNaN(o) && o >= 0 && o <= 7;
  }, t.from = function(o) {
    return t.isValid(o) ? parseInt(o, 10) : void 0;
  }, t.getPenaltyN1 = function(o) {
    const i = o.size;
    let a = 0, c = 0, s = 0, f = null, l = null;
    for (let S = 0; S < i; S++) {
      c = s = 0, f = l = null;
      for (let h = 0; h < i; h++) {
        let v = o.get(S, h);
        v === f ? c++ : (c >= 5 && (a += e.N1 + (c - 5)), f = v, c = 1), v = o.get(h, S), v === l ? s++ : (s >= 5 && (a += e.N1 + (s - 5)), l = v, s = 1);
      }
      c >= 5 && (a += e.N1 + (c - 5)), s >= 5 && (a += e.N1 + (s - 5));
    }
    return a;
  }, t.getPenaltyN2 = function(o) {
    const i = o.size;
    let a = 0;
    for (let c = 0; c < i - 1; c++)
      for (let s = 0; s < i - 1; s++) {
        const f = o.get(c, s) + o.get(c, s + 1) + o.get(c + 1, s) + o.get(c + 1, s + 1);
        (f === 4 || f === 0) && a++;
      }
    return a * e.N2;
  }, t.getPenaltyN3 = function(o) {
    const i = o.size;
    let a = 0, c = 0, s = 0;
    for (let f = 0; f < i; f++) {
      c = s = 0;
      for (let l = 0; l < i; l++)
        c = c << 1 & 2047 | o.get(f, l), l >= 10 && (c === 1488 || c === 93) && a++, s = s << 1 & 2047 | o.get(l, f), l >= 10 && (s === 1488 || s === 93) && a++;
    }
    return a * e.N3;
  }, t.getPenaltyN4 = function(o) {
    let i = 0;
    const a = o.data.length;
    for (let s = 0; s < a; s++)
      i += o.data[s];
    return Math.abs(Math.ceil(i * 100 / a / 5) - 10) * e.N4;
  };
  function n(r, o, i) {
    switch (r) {
      case t.Patterns.PATTERN000:
        return (o + i) % 2 === 0;
      case t.Patterns.PATTERN001:
        return o % 2 === 0;
      case t.Patterns.PATTERN010:
        return i % 3 === 0;
      case t.Patterns.PATTERN011:
        return (o + i) % 3 === 0;
      case t.Patterns.PATTERN100:
        return (Math.floor(o / 2) + Math.floor(i / 3)) % 2 === 0;
      case t.Patterns.PATTERN101:
        return o * i % 2 + o * i % 3 === 0;
      case t.Patterns.PATTERN110:
        return (o * i % 2 + o * i % 3) % 2 === 0;
      case t.Patterns.PATTERN111:
        return (o * i % 3 + (o + i) % 2) % 2 === 0;
      default:
        throw new Error("bad maskPattern:" + r);
    }
  }
  t.applyMask = function(o, i) {
    const a = i.size;
    for (let c = 0; c < a; c++)
      for (let s = 0; s < a; s++)
        i.isReserved(s, c) || i.xor(s, c, n(o, s, c));
  }, t.getBestMask = function(o, i) {
    const a = Object.keys(t.Patterns).length;
    let c = 0, s = 1 / 0;
    for (let f = 0; f < a; f++) {
      i(f), t.applyMask(f, o);
      const l = t.getPenaltyN1(o) + t.getPenaltyN2(o) + t.getPenaltyN3(o) + t.getPenaltyN4(o);
      t.applyMask(f, o), l < s && (s = l, c = f);
    }
    return c;
  };
})(de);
var dt = {};
const V = ft, rt = [
  // L  M  Q  H
  1,
  1,
  1,
  1,
  1,
  1,
  1,
  1,
  1,
  1,
  2,
  2,
  1,
  2,
  2,
  4,
  1,
  2,
  4,
  4,
  2,
  4,
  4,
  4,
  2,
  4,
  6,
  5,
  2,
  4,
  6,
  6,
  2,
  5,
  8,
  8,
  4,
  5,
  8,
  8,
  4,
  5,
  8,
  11,
  4,
  8,
  10,
  11,
  4,
  9,
  12,
  16,
  4,
  9,
  16,
  16,
  6,
  10,
  12,
  18,
  6,
  10,
  17,
  16,
  6,
  11,
  16,
  19,
  6,
  13,
  18,
  21,
  7,
  14,
  21,
  25,
  8,
  16,
  20,
  25,
  8,
  17,
  23,
  25,
  9,
  17,
  23,
  34,
  9,
  18,
  25,
  30,
  10,
  20,
  27,
  32,
  12,
  21,
  29,
  35,
  12,
  23,
  34,
  37,
  12,
  25,
  34,
  40,
  13,
  26,
  35,
  42,
  14,
  28,
  38,
  45,
  15,
  29,
  40,
  48,
  16,
  31,
  43,
  51,
  17,
  33,
  45,
  54,
  18,
  35,
  48,
  57,
  19,
  37,
  51,
  60,
  19,
  38,
  53,
  63,
  20,
  40,
  56,
  66,
  21,
  43,
  59,
  70,
  22,
  45,
  62,
  74,
  24,
  47,
  65,
  77,
  25,
  49,
  68,
  81
], ot = [
  // L  M  Q  H
  7,
  10,
  13,
  17,
  10,
  16,
  22,
  28,
  15,
  26,
  36,
  44,
  20,
  36,
  52,
  64,
  26,
  48,
  72,
  88,
  36,
  64,
  96,
  112,
  40,
  72,
  108,
  130,
  48,
  88,
  132,
  156,
  60,
  110,
  160,
  192,
  72,
  130,
  192,
  224,
  80,
  150,
  224,
  264,
  96,
  176,
  260,
  308,
  104,
  198,
  288,
  352,
  120,
  216,
  320,
  384,
  132,
  240,
  360,
  432,
  144,
  280,
  408,
  480,
  168,
  308,
  448,
  532,
  180,
  338,
  504,
  588,
  196,
  364,
  546,
  650,
  224,
  416,
  600,
  700,
  224,
  442,
  644,
  750,
  252,
  476,
  690,
  816,
  270,
  504,
  750,
  900,
  300,
  560,
  810,
  960,
  312,
  588,
  870,
  1050,
  336,
  644,
  952,
  1110,
  360,
  700,
  1020,
  1200,
  390,
  728,
  1050,
  1260,
  420,
  784,
  1140,
  1350,
  450,
  812,
  1200,
  1440,
  480,
  868,
  1290,
  1530,
  510,
  924,
  1350,
  1620,
  540,
  980,
  1440,
  1710,
  570,
  1036,
  1530,
  1800,
  570,
  1064,
  1590,
  1890,
  600,
  1120,
  1680,
  1980,
  630,
  1204,
  1770,
  2100,
  660,
  1260,
  1860,
  2220,
  720,
  1316,
  1950,
  2310,
  750,
  1372,
  2040,
  2430
];
dt.getBlocksCount = function(e, n) {
  switch (n) {
    case V.L:
      return rt[(e - 1) * 4 + 0];
    case V.M:
      return rt[(e - 1) * 4 + 1];
    case V.Q:
      return rt[(e - 1) * 4 + 2];
    case V.H:
      return rt[(e - 1) * 4 + 3];
    default:
      return;
  }
};
dt.getTotalCodewordsCount = function(e, n) {
  switch (n) {
    case V.L:
      return ot[(e - 1) * 4 + 0];
    case V.M:
      return ot[(e - 1) * 4 + 1];
    case V.Q:
      return ot[(e - 1) * 4 + 2];
    case V.H:
      return ot[(e - 1) * 4 + 3];
    default:
      return;
  }
};
var he = {}, ht = {};
const W = new Uint8Array(512), ct = new Uint8Array(256);
(function() {
  let e = 1;
  for (let n = 0; n < 255; n++)
    W[n] = e, ct[e] = n, e <<= 1, e & 256 && (e ^= 285);
  for (let n = 255; n < 512; n++)
    W[n] = W[n - 255];
})();
ht.log = function(e) {
  if (e < 1)
    throw new Error("log(" + e + ")");
  return ct[e];
};
ht.exp = function(e) {
  return W[e];
};
ht.mul = function(e, n) {
  return e === 0 || n === 0 ? 0 : W[ct[e] + ct[n]];
};
(function(t) {
  const e = ht;
  t.mul = function(r, o) {
    const i = new Uint8Array(r.length + o.length - 1);
    for (let a = 0; a < r.length; a++)
      for (let c = 0; c < o.length; c++)
        i[a + c] ^= e.mul(r[a], o[c]);
    return i;
  }, t.mod = function(r, o) {
    let i = new Uint8Array(r);
    for (; i.length - o.length >= 0; ) {
      const a = i[0];
      for (let s = 0; s < o.length; s++)
        i[s] ^= e.mul(o[s], a);
      let c = 0;
      for (; c < i.length && i[c] === 0; )
        c++;
      i = i.slice(c);
    }
    return i;
  }, t.generateECPolynomial = function(r) {
    let o = new Uint8Array([1]);
    for (let i = 0; i < r; i++)
      o = t.mul(o, new Uint8Array([1, e.exp(i)]));
    return o;
  };
})(he);
const ge = he;
function Lt(t) {
  this.genPoly = void 0, this.degree = t, this.degree && this.initialize(this.degree);
}
Lt.prototype.initialize = function(e) {
  this.degree = e, this.genPoly = ge.generateECPolynomial(this.degree);
};
Lt.prototype.encode = function(e) {
  if (!this.genPoly)
    throw new Error("Encoder not initialized");
  const n = new Uint8Array(e.length + this.degree);
  n.set(e);
  const r = ge.mod(n, this.genPoly), o = this.degree - r.length;
  if (o > 0) {
    const i = new Uint8Array(this.degree);
    return i.set(r, o), i;
  }
  return r;
};
var Qn = Lt, me = {}, H = {}, _t = {};
_t.isValid = function(e) {
  return !isNaN(e) && e >= 1 && e <= 40;
};
var U = {};
const ve = "[0-9]+", Gn = "[A-Z $%*+\\-./:]+";
let J = "(?:[u3000-u303F]|[u3040-u309F]|[u30A0-u30FF]|[uFF00-uFFEF]|[u4E00-u9FAF]|[u2605-u2606]|[u2190-u2195]|u203B|[u2010u2015u2018u2019u2025u2026u201Cu201Du2225u2260]|[u0391-u0451]|[u00A7u00A8u00B1u00B4u00D7u00F7])+";
J = J.replace(/u/g, "\\u");
const Xn = "(?:(?![A-Z0-9 $%*+\\-./:]|" + J + `)(?:.|[\r
]))+`;
U.KANJI = new RegExp(J, "g");
U.BYTE_KANJI = new RegExp("[^A-Z0-9 $%*+\\-./:]+", "g");
U.BYTE = new RegExp(Xn, "g");
U.NUMERIC = new RegExp(ve, "g");
U.ALPHANUMERIC = new RegExp(Gn, "g");
const tr = new RegExp("^" + J + "$"), er = new RegExp("^" + ve + "$"), nr = new RegExp("^[A-Z0-9 $%*+\\-./:]+$");
U.testKanji = function(e) {
  return tr.test(e);
};
U.testNumeric = function(e) {
  return er.test(e);
};
U.testAlphanumeric = function(e) {
  return nr.test(e);
};
(function(t) {
  const e = _t, n = U;
  t.NUMERIC = {
    id: "Numeric",
    bit: 1,
    ccBits: [10, 12, 14]
  }, t.ALPHANUMERIC = {
    id: "Alphanumeric",
    bit: 2,
    ccBits: [9, 11, 13]
  }, t.BYTE = {
    id: "Byte",
    bit: 4,
    ccBits: [8, 16, 16]
  }, t.KANJI = {
    id: "Kanji",
    bit: 8,
    ccBits: [8, 10, 12]
  }, t.MIXED = {
    bit: -1
  }, t.getCharCountIndicator = function(i, a) {
    if (!i.ccBits)
      throw new Error("Invalid mode: " + i);
    if (!e.isValid(a))
      throw new Error("Invalid version: " + a);
    return a >= 1 && a < 10 ? i.ccBits[0] : a < 27 ? i.ccBits[1] : i.ccBits[2];
  }, t.getBestModeForData = function(i) {
    return n.testNumeric(i) ? t.NUMERIC : n.testAlphanumeric(i) ? t.ALPHANUMERIC : n.testKanji(i) ? t.KANJI : t.BYTE;
  }, t.toString = function(i) {
    if (i && i.id)
      return i.id;
    throw new Error("Invalid mode");
  }, t.isValid = function(i) {
    return i && i.bit && i.ccBits;
  };
  function r(o) {
    if (typeof o != "string")
      throw new Error("Param is not a string");
    switch (o.toLowerCase()) {
      case "numeric":
        return t.NUMERIC;
      case "alphanumeric":
        return t.ALPHANUMERIC;
      case "kanji":
        return t.KANJI;
      case "byte":
        return t.BYTE;
      default:
        throw new Error("Unknown mode: " + o);
    }
  }
  t.from = function(i, a) {
    if (t.isValid(i))
      return i;
    try {
      return r(i);
    } catch {
      return a;
    }
  };
})(H);
(function(t) {
  const e = Z, n = dt, r = ft, o = H, i = _t, a = 7973, c = e.getBCHDigit(a);
  function s(h, v, p) {
    for (let u = 1; u <= 40; u++)
      if (v <= t.getCapacity(u, p, h))
        return u;
  }
  function f(h, v) {
    return o.getCharCountIndicator(h, v) + 4;
  }
  function l(h, v) {
    let p = 0;
    return h.forEach(function(u) {
      const E = f(u.mode, v);
      p += E + u.getBitsLength();
    }), p;
  }
  function S(h, v) {
    for (let p = 1; p <= 40; p++)
      if (l(h, p) <= t.getCapacity(p, v, o.MIXED))
        return p;
  }
  t.from = function(v, p) {
    return i.isValid(v) ? parseInt(v, 10) : p;
  }, t.getCapacity = function(v, p, u) {
    if (!i.isValid(v))
      throw new Error("Invalid QR Code version");
    typeof u > "u" && (u = o.BYTE);
    const E = e.getSymbolTotalCodewords(v), g = n.getTotalCodewordsCount(v, p), w = (E - g) * 8;
    if (u === o.MIXED)
      return w;
    const m = w - f(u, v);
    switch (u) {
      case o.NUMERIC:
        return Math.floor(m / 10 * 3);
      case o.ALPHANUMERIC:
        return Math.floor(m / 11 * 2);
      case o.KANJI:
        return Math.floor(m / 13);
      case o.BYTE:
      default:
        return Math.floor(m / 8);
    }
  }, t.getBestVersionForData = function(v, p) {
    let u;
    const E = r.from(p, r.M);
    if (Array.isArray(v)) {
      if (v.length > 1)
        return S(v, E);
      if (v.length === 0)
        return 1;
      u = v[0];
    } else
      u = v;
    return s(u.mode, u.getLength(), E);
  }, t.getEncodedBits = function(v) {
    if (!i.isValid(v) || v < 7)
      throw new Error("Invalid QR Code version");
    let p = v << 12;
    for (; e.getBCHDigit(p) - c >= 0; )
      p ^= a << e.getBCHDigit(p) - c;
    return v << 12 | p;
  };
})(me);
var we = {};
const At = Z, pe = 1335, rr = 21522, jt = At.getBCHDigit(pe);
we.getEncodedBits = function(e, n) {
  const r = e.bit << 3 | n;
  let o = r << 10;
  for (; At.getBCHDigit(o) - jt >= 0; )
    o ^= pe << At.getBCHDigit(o) - jt;
  return (r << 10 | o) ^ rr;
};
var ye = {};
const or = H;
function z(t) {
  this.mode = or.NUMERIC, this.data = t.toString();
}
z.getBitsLength = function(e) {
  return 10 * Math.floor(e / 3) + (e % 3 ? e % 3 * 3 + 1 : 0);
};
z.prototype.getLength = function() {
  return this.data.length;
};
z.prototype.getBitsLength = function() {
  return z.getBitsLength(this.data.length);
};
z.prototype.write = function(e) {
  let n, r, o;
  for (n = 0; n + 3 <= this.data.length; n += 3)
    r = this.data.substr(n, 3), o = parseInt(r, 10), e.put(o, 10);
  const i = this.data.length - n;
  i > 0 && (r = this.data.substr(n), o = parseInt(r, 10), e.put(o, i * 3 + 1));
};
var ir = z;
const ar = H, yt = [
  "0",
  "1",
  "2",
  "3",
  "4",
  "5",
  "6",
  "7",
  "8",
  "9",
  "A",
  "B",
  "C",
  "D",
  "E",
  "F",
  "G",
  "H",
  "I",
  "J",
  "K",
  "L",
  "M",
  "N",
  "O",
  "P",
  "Q",
  "R",
  "S",
  "T",
  "U",
  "V",
  "W",
  "X",
  "Y",
  "Z",
  " ",
  "$",
  "%",
  "*",
  "+",
  "-",
  ".",
  "/",
  ":"
];
function K(t) {
  this.mode = ar.ALPHANUMERIC, this.data = t;
}
K.getBitsLength = function(e) {
  return 11 * Math.floor(e / 2) + 6 * (e % 2);
};
K.prototype.getLength = function() {
  return this.data.length;
};
K.prototype.getBitsLength = function() {
  return K.getBitsLength(this.data.length);
};
K.prototype.write = function(e) {
  let n;
  for (n = 0; n + 2 <= this.data.length; n += 2) {
    let r = yt.indexOf(this.data[n]) * 45;
    r += yt.indexOf(this.data[n + 1]), e.put(r, 11);
  }
  this.data.length % 2 && e.put(yt.indexOf(this.data[n]), 6);
};
var cr = K, sr = function(e) {
  for (var n = [], r = e.length, o = 0; o < r; o++) {
    var i = e.charCodeAt(o);
    if (i >= 55296 && i <= 56319 && r > o + 1) {
      var a = e.charCodeAt(o + 1);
      a >= 56320 && a <= 57343 && (i = (i - 55296) * 1024 + a - 56320 + 65536, o += 1);
    }
    if (i < 128) {
      n.push(i);
      continue;
    }
    if (i < 2048) {
      n.push(i >> 6 | 192), n.push(i & 63 | 128);
      continue;
    }
    if (i < 55296 || i >= 57344 && i < 65536) {
      n.push(i >> 12 | 224), n.push(i >> 6 & 63 | 128), n.push(i & 63 | 128);
      continue;
    }
    if (i >= 65536 && i <= 1114111) {
      n.push(i >> 18 | 240), n.push(i >> 12 & 63 | 128), n.push(i >> 6 & 63 | 128), n.push(i & 63 | 128);
      continue;
    }
    n.push(239, 191, 189);
  }
  return new Uint8Array(n).buffer;
};
const lr = sr, ur = H;
function Y(t) {
  this.mode = ur.BYTE, typeof t == "string" && (t = lr(t)), this.data = new Uint8Array(t);
}
Y.getBitsLength = function(e) {
  return e * 8;
};
Y.prototype.getLength = function() {
  return this.data.length;
};
Y.prototype.getBitsLength = function() {
  return Y.getBitsLength(this.data.length);
};
Y.prototype.write = function(t) {
  for (let e = 0, n = this.data.length; e < n; e++)
    t.put(this.data[e], 8);
};
var fr = Y;
const dr = H, hr = Z;
function $(t) {
  this.mode = dr.KANJI, this.data = t;
}
$.getBitsLength = function(e) {
  return e * 13;
};
$.prototype.getLength = function() {
  return this.data.length;
};
$.prototype.getBitsLength = function() {
  return $.getBitsLength(this.data.length);
};
$.prototype.write = function(t) {
  let e;
  for (e = 0; e < this.data.length; e++) {
    let n = hr.toSJIS(this.data[e]);
    if (n >= 33088 && n <= 40956)
      n -= 33088;
    else if (n >= 57408 && n <= 60351)
      n -= 49472;
    else
      throw new Error(
        "Invalid SJIS character: " + this.data[e] + `
Make sure your charset is UTF-8`
      );
    n = (n >>> 8 & 255) * 192 + (n & 255), t.put(n, 13);
  }
};
var gr = $, Ce = { exports: {} };
(function(t) {
  var e = {
    single_source_shortest_paths: function(n, r, o) {
      var i = {}, a = {};
      a[r] = 0;
      var c = e.PriorityQueue.make();
      c.push(r, 0);
      for (var s, f, l, S, h, v, p, u, E; !c.empty(); ) {
        s = c.pop(), f = s.value, S = s.cost, h = n[f] || {};
        for (l in h)
          h.hasOwnProperty(l) && (v = h[l], p = S + v, u = a[l], E = typeof a[l] > "u", (E || u > p) && (a[l] = p, c.push(l, p), i[l] = f));
      }
      if (typeof o < "u" && typeof a[o] > "u") {
        var g = ["Could not find a path from ", r, " to ", o, "."].join("");
        throw new Error(g);
      }
      return i;
    },
    extract_shortest_path_from_predecessor_list: function(n, r) {
      for (var o = [], i = r; i; )
        o.push(i), n[i], i = n[i];
      return o.reverse(), o;
    },
    find_path: function(n, r, o) {
      var i = e.single_source_shortest_paths(n, r, o);
      return e.extract_shortest_path_from_predecessor_list(
        i,
        o
      );
    },
    /**
     * A very naive priority queue implementation.
     */
    PriorityQueue: {
      make: function(n) {
        var r = e.PriorityQueue, o = {}, i;
        n = n || {};
        for (i in r)
          r.hasOwnProperty(i) && (o[i] = r[i]);
        return o.queue = [], o.sorter = n.sorter || r.default_sorter, o;
      },
      default_sorter: function(n, r) {
        return n.cost - r.cost;
      },
      /**
       * Add a new item to the queue and ensure the highest priority element
       * is at the front of the queue.
       */
      push: function(n, r) {
        var o = { value: n, cost: r };
        this.queue.push(o), this.queue.sort(this.sorter);
      },
      /**
       * Return the highest priority element in the queue.
       */
      pop: function() {
        return this.queue.shift();
      },
      empty: function() {
        return this.queue.length === 0;
      }
    }
  };
  t.exports = e;
})(Ce);
var mr = Ce.exports;
(function(t) {
  const e = H, n = ir, r = cr, o = fr, i = gr, a = U, c = Z, s = mr;
  function f(g) {
    return unescape(encodeURIComponent(g)).length;
  }
  function l(g, w, m) {
    const d = [];
    let y;
    for (; (y = g.exec(m)) !== null; )
      d.push({
        data: y[0],
        index: y.index,
        mode: w,
        length: y[0].length
      });
    return d;
  }
  function S(g) {
    const w = l(a.NUMERIC, e.NUMERIC, g), m = l(a.ALPHANUMERIC, e.ALPHANUMERIC, g);
    let d, y;
    return c.isKanjiModeEnabled() ? (d = l(a.BYTE, e.BYTE, g), y = l(a.KANJI, e.KANJI, g)) : (d = l(a.BYTE_KANJI, e.BYTE, g), y = []), w.concat(m, d, y).sort(function(b, M) {
      return b.index - M.index;
    }).map(function(b) {
      return {
        data: b.data,
        mode: b.mode,
        length: b.length
      };
    });
  }
  function h(g, w) {
    switch (w) {
      case e.NUMERIC:
        return n.getBitsLength(g);
      case e.ALPHANUMERIC:
        return r.getBitsLength(g);
      case e.KANJI:
        return i.getBitsLength(g);
      case e.BYTE:
        return o.getBitsLength(g);
    }
  }
  function v(g) {
    return g.reduce(function(w, m) {
      const d = w.length - 1 >= 0 ? w[w.length - 1] : null;
      return d && d.mode === m.mode ? (w[w.length - 1].data += m.data, w) : (w.push(m), w);
    }, []);
  }
  function p(g) {
    const w = [];
    for (let m = 0; m < g.length; m++) {
      const d = g[m];
      switch (d.mode) {
        case e.NUMERIC:
          w.push([
            d,
            { data: d.data, mode: e.ALPHANUMERIC, length: d.length },
            { data: d.data, mode: e.BYTE, length: d.length }
          ]);
          break;
        case e.ALPHANUMERIC:
          w.push([
            d,
            { data: d.data, mode: e.BYTE, length: d.length }
          ]);
          break;
        case e.KANJI:
          w.push([
            d,
            { data: d.data, mode: e.BYTE, length: f(d.data) }
          ]);
          break;
        case e.BYTE:
          w.push([
            { data: d.data, mode: e.BYTE, length: f(d.data) }
          ]);
      }
    }
    return w;
  }
  function u(g, w) {
    const m = {}, d = { start: {} };
    let y = ["start"];
    for (let B = 0; B < g.length; B++) {
      const b = g[B], M = [];
      for (let A = 0; A < b.length; A++) {
        const N = b[A], k = "" + B + A;
        M.push(k), m[k] = { node: N, lastCount: 0 }, d[k] = {};
        for (let L = 0; L < y.length; L++) {
          const _ = y[L];
          m[_] && m[_].node.mode === N.mode ? (d[_][k] = h(m[_].lastCount + N.length, N.mode) - h(m[_].lastCount, N.mode), m[_].lastCount += N.length) : (m[_] && (m[_].lastCount = N.length), d[_][k] = h(N.length, N.mode) + 4 + e.getCharCountIndicator(N.mode, w));
        }
      }
      y = M;
    }
    for (let B = 0; B < y.length; B++)
      d[y[B]].end = 0;
    return { map: d, table: m };
  }
  function E(g, w) {
    let m;
    const d = e.getBestModeForData(g);
    if (m = e.from(w, d), m !== e.BYTE && m.bit < d.bit)
      throw new Error('"' + g + '" cannot be encoded with mode ' + e.toString(m) + `.
 Suggested mode is: ` + e.toString(d));
    switch (m === e.KANJI && !c.isKanjiModeEnabled() && (m = e.BYTE), m) {
      case e.NUMERIC:
        return new n(g);
      case e.ALPHANUMERIC:
        return new r(g);
      case e.KANJI:
        return new i(g);
      case e.BYTE:
        return new o(g);
    }
  }
  t.fromArray = function(w) {
    return w.reduce(function(m, d) {
      return typeof d == "string" ? m.push(E(d, null)) : d.data && m.push(E(d.data, d.mode)), m;
    }, []);
  }, t.fromString = function(w, m) {
    const d = S(w, c.isKanjiModeEnabled()), y = p(d), B = u(y, m), b = s.find_path(B.map, "start", "end"), M = [];
    for (let A = 1; A < b.length - 1; A++)
      M.push(B.table[b[A]].node);
    return t.fromArray(v(M));
  }, t.rawSplit = function(w) {
    return t.fromArray(
      S(w, c.isKanjiModeEnabled())
    );
  };
})(ye);
const gt = Z, Ct = ft, vr = jn, wr = Jn, pr = ue, yr = fe, Nt = de, Pt = dt, Cr = Qn, st = me, Er = we, Sr = H, Et = ye;
function br(t, e) {
  const n = t.size, r = yr.getPositions(e);
  for (let o = 0; o < r.length; o++) {
    const i = r[o][0], a = r[o][1];
    for (let c = -1; c <= 7; c++)
      if (!(i + c <= -1 || n <= i + c))
        for (let s = -1; s <= 7; s++)
          a + s <= -1 || n <= a + s || (c >= 0 && c <= 6 && (s === 0 || s === 6) || s >= 0 && s <= 6 && (c === 0 || c === 6) || c >= 2 && c <= 4 && s >= 2 && s <= 4 ? t.set(i + c, a + s, !0, !0) : t.set(i + c, a + s, !1, !0));
  }
}
function Br(t) {
  const e = t.size;
  for (let n = 8; n < e - 8; n++) {
    const r = n % 2 === 0;
    t.set(n, 6, r, !0), t.set(6, n, r, !0);
  }
}
function Ir(t, e) {
  const n = pr.getPositions(e);
  for (let r = 0; r < n.length; r++) {
    const o = n[r][0], i = n[r][1];
    for (let a = -2; a <= 2; a++)
      for (let c = -2; c <= 2; c++)
        a === -2 || a === 2 || c === -2 || c === 2 || a === 0 && c === 0 ? t.set(o + a, i + c, !0, !0) : t.set(o + a, i + c, !1, !0);
  }
}
function Mr(t, e) {
  const n = t.size, r = st.getEncodedBits(e);
  let o, i, a;
  for (let c = 0; c < 18; c++)
    o = Math.floor(c / 3), i = c % 3 + n - 8 - 3, a = (r >> c & 1) === 1, t.set(o, i, a, !0), t.set(i, o, a, !0);
}
function St(t, e, n) {
  const r = t.size, o = Er.getEncodedBits(e, n);
  let i, a;
  for (i = 0; i < 15; i++)
    a = (o >> i & 1) === 1, i < 6 ? t.set(i, 8, a, !0) : i < 8 ? t.set(i + 1, 8, a, !0) : t.set(r - 15 + i, 8, a, !0), i < 8 ? t.set(8, r - i - 1, a, !0) : i < 9 ? t.set(8, 15 - i - 1 + 1, a, !0) : t.set(8, 15 - i - 1, a, !0);
  t.set(r - 8, 8, 1, !0);
}
function Ar(t, e) {
  const n = t.size;
  let r = -1, o = n - 1, i = 7, a = 0;
  for (let c = n - 1; c > 0; c -= 2)
    for (c === 6 && c--; ; ) {
      for (let s = 0; s < 2; s++)
        if (!t.isReserved(o, c - s)) {
          let f = !1;
          a < e.length && (f = (e[a] >>> i & 1) === 1), t.set(o, c - s, f), i--, i === -1 && (a++, i = 7);
        }
      if (o += r, o < 0 || n <= o) {
        o -= r, r = -r;
        break;
      }
    }
}
function Nr(t, e, n) {
  const r = new vr();
  n.forEach(function(s) {
    r.put(s.mode.bit, 4), r.put(s.getLength(), Sr.getCharCountIndicator(s.mode, t)), s.write(r);
  });
  const o = gt.getSymbolTotalCodewords(t), i = Pt.getTotalCodewordsCount(t, e), a = (o - i) * 8;
  for (r.getLengthInBits() + 4 <= a && r.put(0, 4); r.getLengthInBits() % 8 !== 0; )
    r.putBit(0);
  const c = (a - r.getLengthInBits()) / 8;
  for (let s = 0; s < c; s++)
    r.put(s % 2 ? 17 : 236, 8);
  return Pr(r, t, e);
}
function Pr(t, e, n) {
  const r = gt.getSymbolTotalCodewords(e), o = Pt.getTotalCodewordsCount(e, n), i = r - o, a = Pt.getBlocksCount(e, n), c = r % a, s = a - c, f = Math.floor(r / a), l = Math.floor(i / a), S = l + 1, h = f - l, v = new Cr(h);
  let p = 0;
  const u = new Array(a), E = new Array(a);
  let g = 0;
  const w = new Uint8Array(t.buffer);
  for (let b = 0; b < a; b++) {
    const M = b < s ? l : S;
    u[b] = w.slice(p, p + M), E[b] = v.encode(u[b]), p += M, g = Math.max(g, M);
  }
  const m = new Uint8Array(r);
  let d = 0, y, B;
  for (y = 0; y < g; y++)
    for (B = 0; B < a; B++)
      y < u[B].length && (m[d++] = u[B][y]);
  for (y = 0; y < h; y++)
    for (B = 0; B < a; B++)
      m[d++] = E[B][y];
  return m;
}
function Tr(t, e, n, r) {
  let o;
  if (Array.isArray(t))
    o = Et.fromArray(t);
  else if (typeof t == "string") {
    let f = e;
    if (!f) {
      const l = Et.rawSplit(t);
      f = st.getBestVersionForData(l, n);
    }
    o = Et.fromString(t, f || 40);
  } else
    throw new Error("Invalid data");
  const i = st.getBestVersionForData(o, n);
  if (!i)
    throw new Error("The amount of data is too big to be stored in a QR Code");
  if (!e)
    e = i;
  else if (e < i)
    throw new Error(
      `
The chosen QR Code version cannot contain this amount of data.
Minimum version required to store current data is: ` + i + `.
`
    );
  const a = Nr(e, n, o), c = gt.getSymbolSize(e), s = new wr(c);
  return br(s, e), Br(s), Ir(s, e), St(s, n, 0), e >= 7 && Mr(s, e), Ar(s, a), isNaN(r) && (r = Nt.getBestMask(
    s,
    St.bind(null, s, n)
  )), Nt.applyMask(r, s), St(s, n, r), {
    modules: s,
    version: e,
    errorCorrectionLevel: n,
    maskPattern: r,
    segments: o
  };
}
se.create = function(e, n) {
  if (typeof e > "u" || e === "")
    throw new Error("No input text");
  let r = Ct.M, o, i;
  return typeof n < "u" && (r = Ct.from(n.errorCorrectionLevel, Ct.M), o = st.from(n.version), i = Nt.from(n.maskPattern), n.toSJISFunc && gt.setToSJISFunction(n.toSJISFunc)), Tr(e, o, r, i);
};
var Ee = {}, Zt = {};
(function(t) {
  function e(n) {
    if (typeof n == "number" && (n = n.toString()), typeof n != "string")
      throw new Error("Color should be defined as hex string");
    let r = n.slice().replace("#", "").split("");
    if (r.length < 3 || r.length === 5 || r.length > 8)
      throw new Error("Invalid hex color: " + n);
    (r.length === 3 || r.length === 4) && (r = Array.prototype.concat.apply([], r.map(function(i) {
      return [i, i];
    }))), r.length === 6 && r.push("F", "F");
    const o = parseInt(r.join(""), 16);
    return {
      r: o >> 24 & 255,
      g: o >> 16 & 255,
      b: o >> 8 & 255,
      a: o & 255,
      hex: "#" + r.slice(0, 6).join("")
    };
  }
  t.getOptions = function(r) {
    r || (r = {}), r.color || (r.color = {});
    const o = typeof r.margin > "u" || r.margin === null || r.margin < 0 ? 4 : r.margin, i = r.width && r.width >= 21 ? r.width : void 0, a = r.scale || 4;
    return {
      width: i,
      scale: i ? 4 : a,
      margin: o,
      color: {
        dark: e(r.color.dark || "#000000ff"),
        light: e(r.color.light || "#ffffffff")
      },
      type: r.type,
      rendererOpts: r.rendererOpts || {}
    };
  }, t.getScale = function(r, o) {
    return o.width && o.width >= r + o.margin * 2 ? o.width / (r + o.margin * 2) : o.scale;
  }, t.getImageWidth = function(r, o) {
    const i = t.getScale(r, o);
    return Math.floor((r + o.margin * 2) * i);
  }, t.qrToImageData = function(r, o, i) {
    const a = o.modules.size, c = o.modules.data, s = t.getScale(a, i), f = Math.floor((a + i.margin * 2) * s), l = i.margin * s, S = [i.color.light, i.color.dark];
    for (let h = 0; h < f; h++)
      for (let v = 0; v < f; v++) {
        let p = (h * f + v) * 4, u = i.color.light;
        if (h >= l && v >= l && h < f - l && v < f - l) {
          const E = Math.floor((h - l) / s), g = Math.floor((v - l) / s);
          u = S[c[E * a + g] ? 1 : 0];
        }
        r[p++] = u.r, r[p++] = u.g, r[p++] = u.b, r[p] = u.a;
      }
  };
})(Zt);
(function(t) {
  const e = Zt;
  function n(o, i, a) {
    o.clearRect(0, 0, i.width, i.height), i.style || (i.style = {}), i.height = a, i.width = a, i.style.height = a + "px", i.style.width = a + "px";
  }
  function r() {
    try {
      return document.createElement("canvas");
    } catch {
      throw new Error("You need to specify a canvas element");
    }
  }
  t.render = function(i, a, c) {
    let s = c, f = a;
    typeof s > "u" && (!a || !a.getContext) && (s = a, a = void 0), a || (f = r()), s = e.getOptions(s);
    const l = e.getImageWidth(i.modules.size, s), S = f.getContext("2d"), h = S.createImageData(l, l);
    return e.qrToImageData(h.data, i, s), n(S, f, l), S.putImageData(h, 0, 0), f;
  }, t.renderToDataURL = function(i, a, c) {
    let s = c;
    typeof s > "u" && (!a || !a.getContext) && (s = a, a = void 0), s || (s = {});
    const f = t.render(i, a, s), l = s.type || "image/png", S = s.rendererOpts || {};
    return f.toDataURL(l, S.quality);
  };
})(Ee);
var Se = {};
const Rr = Zt;
function Jt(t, e) {
  const n = t.a / 255, r = e + '="' + t.hex + '"';
  return n < 1 ? r + " " + e + '-opacity="' + n.toFixed(2).slice(1) + '"' : r;
}
function bt(t, e, n) {
  let r = t + e;
  return typeof n < "u" && (r += " " + n), r;
}
function kr(t, e, n) {
  let r = "", o = 0, i = !1, a = 0;
  for (let c = 0; c < t.length; c++) {
    const s = Math.floor(c % e), f = Math.floor(c / e);
    !s && !i && (i = !0), t[c] ? (a++, c > 0 && s > 0 && t[c - 1] || (r += i ? bt("M", s + n, 0.5 + f + n) : bt("m", o, 0), o = 0, i = !1), s + 1 < e && t[c + 1] || (r += bt("h", a), a = 0)) : o++;
  }
  return r;
}
Se.render = function(e, n, r) {
  const o = Rr.getOptions(n), i = e.modules.size, a = e.modules.data, c = i + o.margin * 2, s = o.color.light.a ? "<path " + Jt(o.color.light, "fill") + ' d="M0 0h' + c + "v" + c + 'H0z"/>' : "", f = "<path " + Jt(o.color.dark, "stroke") + ' d="' + kr(a, i, o.margin) + '"/>', l = 'viewBox="0 0 ' + c + " " + c + '"', h = '<svg xmlns="http://www.w3.org/2000/svg" ' + (o.width ? 'width="' + o.width + '" height="' + o.width + '" ' : "") + l + ' shape-rendering="crispEdges">' + s + f + `</svg>
`;
  return typeof r == "function" && r(null, h), h;
};
const Lr = $n, Tt = se, be = Ee, _r = Se;
function Ft(t, e, n, r, o) {
  const i = [].slice.call(arguments, 1), a = i.length, c = typeof i[a - 1] == "function";
  if (!c && !Lr())
    throw new Error("Callback required as last argument");
  if (c) {
    if (a < 2)
      throw new Error("Too few arguments provided");
    a === 2 ? (o = n, n = e, e = r = void 0) : a === 3 && (e.getContext && typeof o > "u" ? (o = r, r = void 0) : (o = r, r = n, n = e, e = void 0));
  } else {
    if (a < 1)
      throw new Error("Too few arguments provided");
    return a === 1 ? (n = e, e = r = void 0) : a === 2 && !e.getContext && (r = n, n = e, e = void 0), new Promise(function(s, f) {
      try {
        const l = Tt.create(n, r);
        s(t(l, e, r));
      } catch (l) {
        f(l);
      }
    });
  }
  try {
    const s = Tt.create(n, r);
    o(null, t(s, e, r));
  } catch (s) {
    o(s);
  }
}
Q.create = Tt.create;
Q.toCanvas = Ft.bind(null, be.render);
Q.toDataURL = Ft.bind(null, be.renderToDataURL);
Q.toString = Ft.bind(null, function(t, e, n) {
  return _r.render(t, n);
});
const Zr = (t, e) => {
  const n = Array.prototype.slice.call(
    Q.create(t, { errorCorrectionLevel: e }).modules.data,
    0
  ), r = Math.sqrt(n.length);
  return n.reduce(
    (o, i, a) => (a % r === 0 ? o.push([i]) : o[o.length - 1].push(i)) && o,
    []
  );
};
function Fr({
  ecl: t = "H",
  logoMargin: e = 10,
  logoSize: n = 50,
  size: r = 200,
  uri: o
}) {
  const a = r - parseInt("20", 10) * 2, c = Ae(() => {
    const l = [], S = Zr(o, t), h = a / S.length;
    [
      { x: 0, y: 0 },
      { x: 1, y: 0 },
      { x: 0, y: 1 }
    ].forEach(({ x: g, y: w }) => {
      const m = (S.length - 7) * h * g, d = (S.length - 7) * h * w;
      for (let y = 0; y < 3; y++)
        l.push(
          /* @__PURE__ */ C(
            "rect",
            {
              fill: y % 2 !== 0 ? "white" : "black",
              height: h * (7 - y * 2),
              rx: (y - 2) * -5 + (y === 0 ? 2 : 0),
              ry: (y - 2) * -5 + (y === 0 ? 2 : 0),
              width: h * (7 - y * 2),
              x: m + h * y,
              y: d + h * y
            },
            `${y}-${g}-${w}`
          )
        );
    });
    const p = Math.floor((n + e * 2) / h), u = S.length / 2 - p / 2, E = S.length / 2 + p / 2 - 1;
    return S.forEach((g, w) => {
      g.forEach((m, d) => {
        S[w][d] && (w < 7 && d < 7 || w > S.length - 8 && d < 7 || w < 7 && d > S.length - 8 || w > u && w < E && d > u && d < E || l.push(
          /* @__PURE__ */ C(
            "circle",
            {
              cx: w * h + h / 2,
              cy: d * h + h / 2,
              fill: "black",
              r: h / 3
            },
            `circle-${w}-${d}`
          )
        ));
      });
    }), l;
  }, [t, n, e, a, o]), s = a / 2 - n / 2, f = n + e * 2;
  return /* @__PURE__ */ C("div", { className: Ve, children: /* @__PURE__ */ T("div", { className: He, children: [
    /* @__PURE__ */ C(
      "div",
      {
        className: Ue,
        style: {
          top: s,
          width: a
        },
        children: /* @__PURE__ */ C(Xt, { fill: "purple", height: n })
      }
    ),
    /* @__PURE__ */ T("svg", { height: a, style: { all: "revert" }, width: a, children: [
      /* @__PURE__ */ C("title", { children: "QR Code" }),
      /* @__PURE__ */ T("defs", { children: [
        /* @__PURE__ */ C("clipPath", { id: "clip-wrapper", children: /* @__PURE__ */ C("rect", { height: f, width: f }) }),
        /* @__PURE__ */ C("clipPath", { id: "clip-logo", children: /* @__PURE__ */ C("rect", { height: n, width: n }) })
      ] }),
      /* @__PURE__ */ C("rect", { fill: "transparent", height: a, width: a }),
      c
    ] })
  ] }) });
}
function Dr({
  open: t,
  onClose: e,
  url: n,
  isError: r,
  error: o
}) {
  return /* @__PURE__ */ C(Kn, { open: t, titleId: "Sign In With Farcaster", onClose: e, children: /* @__PURE__ */ C("div", { className: "fc-authkit-qrcode-dialog", children: /* @__PURE__ */ T("div", { className: Yn, children: [
    /* @__PURE__ */ C(
      Bt,
      {
        kind: "reset",
        onClick: e,
        style: { position: "absolute", top: 19, right: 13 },
        children: /* @__PURE__ */ C(
          "svg",
          {
            xmlns: "http://www.w3.org/2000/svg",
            width: 18,
            height: 18,
            fill: "none",
            children: /* @__PURE__ */ C(
              "path",
              {
                fill: "rgba(0,0,0,0.5)",
                fillRule: "evenodd",
                d: "M1.15 1.15a.937.937 0 0 1 1.325 0L9 7.674l6.525-6.524a.937.937 0 1 1 1.325 1.325L10.326 9l6.524 6.525a.937.937 0 0 1-1.325 1.325L9 10.326 2.475 16.85a.938.938 0 0 1-1.325-1.325L7.674 9 1.15 2.475a.937.937 0 0 1 0-1.325Z",
                clipRule: "evenodd"
              }
            )
          }
        )
      }
    ),
    r ? /* @__PURE__ */ T(j, { children: [
      /* @__PURE__ */ C("div", { className: $t, children: "Error" }),
      /* @__PURE__ */ C("div", { className: Yt, children: (o == null ? void 0 : o.message) ?? "Unknown error, please try again." })
    ] }) : /* @__PURE__ */ T(j, { children: [
      /* @__PURE__ */ C("div", { className: $t, children: "Sign in with Farcaster" }),
      /* @__PURE__ */ C("div", { className: Yt, children: "Scan with your phone's camera to continue." }),
      /* @__PURE__ */ C(
        "div",
        {
          style: {
            display: "flex",
            justifyContent: "center",
            marginTop: 24,
            marginBottom: 12
          },
          children: /* @__PURE__ */ C(Fr, { uri: n, size: 264, logoSize: 22, logoMargin: 12 })
        }
      ),
      /* @__PURE__ */ C("div", { style: { display: "flex", justifyContent: "center" }, children: /* @__PURE__ */ T(
        Bt,
        {
          kind: "tertiary",
          style: {
            display: "flex",
            alignItems: "center",
            fontWeight: 500
          },
          onClick: () => {
            window.location.href = n;
          },
          children: [
            /* @__PURE__ */ C(
              "svg",
              {
                xmlns: "http://www.w3.org/2000/svg",
                width: 12,
                height: 18,
                fill: "none",
                children: /* @__PURE__ */ C(
                  "path",
                  {
                    fill: "#7C65C1",
                    fillRule: "evenodd",
                    d: "M0 3a3 3 0 0 1 3-3h6a3 3 0 0 1 3 3v12a3 3 0 0 1-3 3H3a3 3 0 0 1-3-3V3Zm4-1.5v.75c0 .414.336.75.75.75h2.5A.75.75 0 0 0 8 2.25V1.5h1A1.5 1.5 0 0 1 10.5 3v12A1.5 1.5 0 0 1 9 16.5H3A1.5 1.5 0 0 1 1.5 15V3A1.5 1.5 0 0 1 3 1.5h1Z",
                    clipRule: "evenodd"
                  }
                )
              }
            ),
            /* @__PURE__ */ C("span", { style: { marginLeft: 9 }, children: "I'm using my phone →" })
          ]
        }
      ) })
    ] })
  ] }) }) });
}
var Ur = "_8rz8bk0";
function Vr({
  debug: t,
  hideSignOut: e,
  onSignOut: n,
  ...r
}) {
  const { onSuccess: o, onStatusResponse: i, onError: a } = r, c = R(
    (L) => {
      o == null || o(L);
    },
    [o]
  ), s = R(
    (L) => {
      i == null || i(L);
    },
    [i]
  ), f = R(
    (L) => {
      a == null || a(L);
    },
    [a]
  ), l = R(() => {
    n == null || n();
  }, [n]), S = Fe({
    ...r,
    onSuccess: c,
    onStatusResponse: s,
    onError: f
  }), {
    signIn: h,
    signOut: v,
    connect: p,
    reconnect: u,
    isSuccess: E,
    isError: g,
    error: w,
    channelToken: m,
    url: d,
    data: y,
    validSignature: B
  } = S, b = R(() => {
    A(!1), v(), l();
  }, [v, l]), [M, A] = P(!1), N = R(() => {
    g && u(), A(!0), h(), d && Mt() && (window.location.href = d);
  }, [g, u, h, d]), k = E && B;
  return F(() => {
    m || p();
  }, [m, p]), /* @__PURE__ */ T("div", { className: "fc-authkit-signin-button", children: [
    k ? /* @__PURE__ */ C(
      Ge,
      {
        userData: y,
        signOut: b,
        hideSignOut: !!e
      }
    ) : /* @__PURE__ */ T(j, { children: [
      /* @__PURE__ */ C(Ke, { onClick: N, label: "Sign in" }),
      d && /* @__PURE__ */ C(
        Dr,
        {
          open: M && !Mt(),
          onClose: () => A(!1),
          url: d,
          isError: g,
          error: w
        }
      )
    ] }),
    t && /* @__PURE__ */ C("div", { className: Ur, children: /* @__PURE__ */ C("pre", { children: JSON.stringify(S, null, 2) }) })
  ] });
}
function Hr() {
  const { isAuthenticated: t, profile: e } = q();
  return { isAuthenticated: t, profile: e };
}
function $r() {
  return /* @__PURE__ */ T(Te, { config: {
    rpcUrl: "https://mainnet.optimism.io"
  }, children: [
    /* @__PURE__ */ C("div", { style: { position: "fixed", top: 12, right: 12, zIndex: 1e3 }, children: /* @__PURE__ */ C(
      Vr,
      {
        nonce: "abcd1234",
        requestId: "some-unique-request-id",
        timeout: 2e4,
        onError: (e) => console.error("error callback:", e),
        onSuccess: (e) => console.log("success callback:", e),
        onStatusResponse: (e) => console.log("status callback:", e),
        onSignOut: () => console.log("sign out callback"),
        debug: !0
      }
    ) }),
    /* @__PURE__ */ C(xr, {})
  ] });
}
function xr() {
  const {
    isAuthenticated: t,
    profile: { fid: e, bio: n, displayName: r, custody: o }
  } = Hr();
  return /* @__PURE__ */ C("div", { style: { fontFamily: "sans-serif" }, children: t && /* @__PURE__ */ T("div", { children: [
    e && /* @__PURE__ */ T("p", { children: [
      /* @__PURE__ */ C("strong", { children: "FID:" }),
      " ",
      e
    ] }),
    r && /* @__PURE__ */ T("p", { children: [
      /* @__PURE__ */ C("strong", { children: "Display name:" }),
      " ",
      r
    ] }),
    n && /* @__PURE__ */ T("p", { children: [
      /* @__PURE__ */ C("strong", { children: "Bio:" }),
      " ",
      n
    ] }),
    o && /* @__PURE__ */ T("p", { children: [
      /* @__PURE__ */ C("strong", { children: "Custody address:" }),
      " ",
      o
    ] })
  ] }) });
}
function Wr() {
  const { signInMessage: t } = q();
  return t;
}
export {
  Te as AuthKitProvider,
  $r as Demo,
  Fr as QRCode,
  Vr as SignInButton,
  lt as useAppClient,
  Re as useCreateChannel,
  Hr as useProfile,
  Fe as useSignIn,
  Wr as useSignInMessage,
  ke as useVerifySignInMessage,
  _e as useWatchStatus
};
